Library/Homebrew/cmd/update-bash.sh

```diff
@@ -289,10 +301,16 @@ EOS
   # this procedure will be removed in the future if it seems unnecessary
   rename_taps_dir_if_necessary
 
+  DIRS=("$HOMEBREW_LIBRARY"/Taps/*/*)
+  if [[ -z "$SKIP_UPDATE_HOMEBREW_REPOSITOR" ]]
+  then
+    DIRS=("$HOMEBREW_REPOSITORY" "${DIRS[@]}")
+  fi
```

Comment:
Personally I think using new data structures provided only by Bash and not other POSIX implementations is a step too far. I don't think Bash arrays are widely used or understood. If the majority disagrees: so be it but I do think it's worth aiming for code that's easily understood over almost any other criteria as otherwise bugs creep in when the original intent is not understood fully.